pydis_site/apps/content/resources/guides/pydis-guides/contributing/bot.md

```diff
@@ -37,21 +36,115 @@ If your bot fails to start with a `PrivilegedIntentsRequired` exception, this in
+**Congratulations**, you have finished the configuration & can now start [running your bot](#run-it)
+
+#### Manual configuration
+##### .env.server
+All server configuration values are saved in a file called `.env.server`, which needs to be at the root directory of the cloned code.
 This file contains the various configurations we use to make the bot run on the Python Discord server, such as channel and role IDs, and the emojis it works with.
 It also contains configurations such as how long it takes for a help channel to time out, and how many messages a user needs to voice-verify.
 
-To run the bot in your test server, you will need to override some of those configurations.
-Create and open a new file in the directory called `config.yml`. Alternatively, copy the `config-default.yml` file and rename the copy to `config.yml`.
-The bot will first look at the items in `config.yml`, and will fall back to `config-default.yml` only if necessary. Note that you don't have to specify all items in `config.yml`, just the ones you want to override such as channel IDs.
+If you decided to use the bootstrapping script, you'll find that this file has already been created (which we recommend),
+otherwise you'll need to create it manually.
+
+If you decide to set the configuration values manually, you will **only** need to set the values for the channels, roles, categories, etc.
+that are used by the component you are developing.
+
+
+Let's take an example where we suppose we'll only be testing a feature that needs the `announcements` channel.
+
+`constants.py`
+
+```py
+
+class EnvConfig:
+    # Defines from where & how Pydantic will be looking for env variables
+    ...
+
+class _Channels(EnvConfig):
+
+    EnvConfig.Config.env_prefix = "channels_"
+
+    announcements = 1079790565794779156
+    changelog = 1077877318564991006
+
+# Instantiate the class & load the configuration
+Channels = _Channels()
+```
+
+`.env.server` file
+
+```text
+channels_announcements=1077875228002234398
+```
+
+When you launch your bot, `pydantic` will load up the server constants from the `.env.server` file if they exist.
+
+Each constants class will define its own prefix, which will make `pydantic` look for variables that will look like `{{env_prefix}}{{attribute_name}}` in the environment files
+
+In our example, this will imply that pydantic will look for both `channels_announcements` and `channels_changelog` in the `.env.server` file.
+
+As you can see here, only `channels_announcements` has been defined in the `.env.server` file since it's the only one needed, which will tell `pydantic`
+to use the value **1077875228002234398** for the `announcements` attribute instead of the default **1079790565794779156**, and use the default value for the `changelog` attribute
+
+```python
+>>> Channels.announcements
+1077875228002234398
+>>> Channels.changelong
+1077877318564991006
+```
 
 See [here](../obtaining-discord-ids) for help with obtaining Discord IDs.
 
 <div class="card">
```

Comment:
@swfarnsworth and I discussed this before, I also had doubts about putting it here, but I also thought of people not being able to understand what's going on, especially new contribs.
If you have a strong opinion about removing it, I wouldn't mind doing it at all.

Reply:
Yes I'm in favor of still explaining how the .env.server is structured and how to modify specific constants, I just don't think the huge card is needed anymore.

I tried used the script yesterday, and it's very simple. In the planning stages I talked about needing a transition script from the old yaml config to the new one, but honestly it's much simpler to just run the script, even if it means starting a new server. I don't even want to suggest setting the entire .env.server file manually as a viable option.

Reply:
I agree with you here.

However, I wasn't talking about setting the entire env.server manually, but mostly about how would you go on about setting some variable if you ever need it.

To us it is very obvious because we discussed, planned, implemented and tested it, but idk how it would look like from the perspective of someone who has no clue about it.

Reply:
No no, like I said I'm in favor of keeping it. I'm talking about the massive collapsible card.

Reply:
Ah ok, I understood something else completely.
I'll also add small explanation of nested objects in that case, which will cover the antispam & rules cases

Reply:
You mean the parts I'm deleting in my PR? :P